File: lib/models/inventory/damage_item_model.dart
class DamageItem {
  final int? itemId;
  final String itemCode;
  final String itemName;
  final String unit;
  final int qty;
  final double rate;
  final String remarks;

  DamageItem({
    this.itemId,
    required this.itemCode,
    required this.itemName,
    required this.unit,
    required this.qty,
    required this.rate,
    required this.remarks,
  });

  double get amount => qty * rate;

  Map<String, dynamic> toJson() {
    return {
      'item_id': itemId,
      'item_code': itemCode,
      'item_name': itemName,
      'unit': unit,
      'qty': qty,
      'rate': rate,
      'remarks': remarks,
    };
  }
}

File: backend/models/property/damageItem.model.js
module.exports = (sequelize, DataTypes) => {
    const DamageItem = sequelize.define('damage_items', {
        damage_id: DataTypes.INTEGER,
        item_id: DataTypes.INTEGER,
        qty: DataTypes.DECIMAL(12, 2),
        rate: DataTypes.DECIMAL(12, 2),
        remarks: DataTypes.TEXT,
        amount: DataTypes.DECIMAL(12, 2)
    }, {
        tableName: 'damage_items',
        timestamps: false
    });

    DamageItem.associate = (models) => {
        DamageItem.belongsTo(models.damage_headers, {
            foreignKey: 'damage_id',
            as: 'header'
        });
        DamageItem.belongsTo(models.item_master, {
            foreignKey: 'item_id',
            as: 'item'
        });
    };

    return DamageItem;
};
